recovery/device.go

```diff
@@ -115,18 +114,16 @@ func (d *DiskDevice) CreateLUKSPartition(size uint64, label string, keyfile stri
 	// FIXME: determine partition name in a civilized way
 	partdev := d.partDev(4)
 
-	// Set up LUKS device
-	logger.Noticef("Create LUKS keyfile")
-	buffer := make([]byte, keyfileSize)
-	rand.Read(buffer)
-	if err := ioutil.WriteFile(keyfile, buffer, 0400); err != nil {
-		return fmt.Errorf("cannot create keyfile %s: %s", keyfile, err)
-	}
-
-	// Don't remove this delay, prevents kernel crash
+	// Don't remove this delay, it prevents a kernel crash
 	// see https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1835279
 	time.Sleep(1 * time.Second)
 
+	// Create a temporary unsealed keyfile on a RAM-backed filesystem
+	keyfile := "/run/tmpkeyfile"
+	if err := ioutil.WriteFile(keyfile, keyBuffer, 0400); err != nil {
+		return fmt.Errorf("cannot create keyfile: %s", err)
+	}
```

Comment:
If you use a real file this would be a good place to defer removal of it.

Reply:
Temporary keyfiles removed, now passing key data to stdin.